ledger-service/http-json/src/main/scala/com/digitalasset/http/EndpointsCompanion.scala

```diff
@@ -175,12 +298,41 @@ object EndpointsCompanion {
 
   private[http] def format(a: JsValue): ByteString = ByteString(a.compactPrint)
 
-  private[http] def decodeAndParsePayload[A](jwt: Jwt, decodeJwt: ValidateJwt)(implicit
+  private[http] def legacyDecodeAndParsePayload[A](jwt: Jwt, decodeJwt: ValidateJwt)(implicit
       parse: ParsePayload[A]
-  ): Unauthorized \/ (Jwt, A) = {
+  ) =
     for {
       a <- decodeJwt(jwt): Unauthorized \/ DecodedJwt[String]
       p <- parse.parsePayload(a)
     } yield (jwt, p)
+
+  private[http] def decodeAndParsePayload[A](
+      jwt: Jwt,
+      decodeJwt: ValidateJwt,
+      userManagementClient: UserManagementClient,
+      ledgerIdentityClient: LedgerIdentityClient,
+  )(implicit
+      lc: LoggingContextOf[InstanceUUID with RequestID],
+      legacyParse: ParsePayload[A],
```

Comment:
maybe consider sticking to the naming used in the ledger API. Calling it legacy in one place and custom in another place is just asking for confusion.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/12075/commits/e67e117848426849692653fd8992de8d28895c77, I decided to stick with "custom" because it's the name used on the server.